pkg/datapath/iptables/iptables.go

```diff
@@ -840,9 +895,13 @@ func (m *IptablesManager) GetProxyPort(name string) uint16 {
 		return 0
 	}
 
-	re := regexp.MustCompile(name + ".*TPROXY redirect 0.0.0.0:([1-9][0-9]*) mark")
-	str := re.FindString(string(res))
-	portStr := re.ReplaceAllString(str, "$1")
+	re := regexp.MustCompile(name + ".*TPROXY redirect (0.0.0.0|::):([1-9][0-9]*) mark")
+	strs := re.FindAllString(string(res), -1)
+	if len(strs) == 0 {
+		return 0
+	}
```

Comment:
What's the impact or need for this particular change in this commit, in context of the rest of the changes proposed here? Is it possible to put this in a separate commit and describe why this is necessary, or is it integral to the overall approach?

Reply:
Previously only IPv4 address format (`0.0.0.0`) was detected, with this change also IPv6 address format (`::`) is supported. This is a separate issue, but the rest here is integral to the new approach:

Instead of finding the first match and getting the proxy port from there we need to find the last (== latest) match and get the proxy port from there. This is significant if/when there are multiple TPROXY rules for the same proxy (e.g., DNS), some old and the last one new. This could happen if on restart an explicit proxy port is configured and we failed to clean up the old rule from the chain. Without this change the next restart would then try to use the stale port of the 1st match. We do clean up the old rules, but if Cilium gets killed right before the cleanup we may find multiple rules in the chain for the same proxy.

New rules are added with `-A`, so they are always last in the chain.

Reply:
This is quite nuanced, do you think it should be encoded as a comment or in the git logs as well?